cloudprovider/cloudprovider.go

```diff
@@ -100,7 +100,11 @@ func GetInstance(cloud Interface, IP string) (instance *types.Instance, err erro
 	runningMutex.Lock()
 	running[IP] = time.Now()
 	runningMutex.Unlock()
-	defer func() { delete(running, IP) }()
+	defer func() {
```

Comment:
> reads are safe usually

From the "Advice" section of https://golang.org/ref/mem :

> Programs that modify data being simultaneously accessed by multiple goroutines must serialize such access.

Access to shared mutable state must be always synchronized for **both** reads and writes. Otherwise "Bad Things will happen" :) I can dig up some material if you are interested in gory details.